consensus/src/dag/order_rule.rs

```diff
@@ -196,4 +193,25 @@ impl OrderRule {
             error!("Failed to send ordered nodes {:?}", e);
         }
     }
+
+    /// Check if this node can trigger anchors to be ordered
+    pub fn process_new_node(&mut self, node_metadata: &NodeMetadata) {
+        let round = node_metadata.round();
+        // If the node comes from the proposal round in the current instance, it can't trigger any ordering
+        if round <= self.lowest_unordered_anchor_round
+            || Self::check_parity(round, self.lowest_unordered_anchor_round)
+        {
+            return;
+        }
+        // This node's votes can trigger an anchor from previous round to be ordered.
+        let start_round = round - 1;
+        self.check_ordering_between(start_round, round)
+    }
+
+    /// Check the whole dag to see if anything can be ordered.
+    pub fn process_all(&mut self) {
+        let start_round = self.lowest_unordered_anchor_round;
+        let round = self.dag.read().highest_round();
```

Comment:
Will this drop the lock before calling `check_ordering_between`? I usually wrap with curly braces because I was never sure.

Reply:
the guard is dropped inline with chaining operations like this